### ISSUE TITLE
Trouble with setup

### ISSUE THREAD

**Issue by altruity** [tinloof/medusa-dtc-starter-munchies#45]
First off, huge respect for the brilliant work on this project 🙌  Super impressed with every component from the Figma to the CMS to this starter. 

I ran into a few issues during setup and am wondering if others may be experiencing this. Perhaps there are some missing steps in the README? 

https://github.com/tinloof/medusa-dtc-starter-munchies?tab=readme-ov-file#medusa-setup

```
To set up Medusa for your project:
Once project is deployed...
```

Also, running `pnpm install` from the root, gives this error:
` ERROR  This project is configured to use yarn`
It does seem so, checking `package.json`:
https://github.com/tinloof/medusa-dtc-starter-munchies/blob/main/backend/package.json#L60

Finally, running a build, after installing with yarn in the backend dir, fails with stuff like:
```
src/api/admin/custom/route.ts:1:10 - error TS2305: Module '"@medusajs/medusa"' has no exported member 'MedusaRequest'.
1 import { MedusaRequest, MedusaResponse } from "@medusajs/medusa";
```

This leads me to believe that I'm on the wrong track. There must be some sequence of commands that I am not running to ensure a quick and easy install. Amazing Tinloof team, any guidance is highly appreciated! 🙏  Afterwords, I'd be happy to create a PR with complete set up documentation, if desired.

**Comment by BloodFalka** [comment]
Almost everything in the README file is outdated. I ran the backend and storefront separately (`yarn dev `for the backend and `pnpm dev` for the storefront). In addition, many .env variables need to be configured, for example, the backend will not start without the stripe and sendgrid api keys. also, be sure to run the `medusa db:migrate` command before the first launch of the backend. The README file is definitely worth updating, I spent the whole night running the project, but it was worth it. I still can't build the frontend though, yarn/pnpm build always fails with an error

**Comment by DETDubai** [comment]
As for this error, it is a typescript error and because of this, the build fails.

```
src/api/admin/custom/route.ts:1:10 - error TS2305: Module '"@medusajs/medusa"' has no exported member 'MedusaRequest'.
1 import { MedusaRequest, MedusaResponse } from "@medusajs/medusa";
```

I didn't really understand what the problem was and just replaced all `import { MedusaRequest, MedusaResponse } from "@medusajs/medusa";` with `import { Request, Response } from "express";`

Also, why do you need to make a build now? First, run the application through `yarn dev` and check that everything is configured correctly

**Comment by BloodFalka** [comment]
As for this error, it is a typescript error and because of this, the build fails.

```
src/api/admin/custom/route.ts:1:10 - error TS2305: Module '"@medusajs/medusa"' has no exported member 'MedusaRequest'.
1 import { MedusaRequest, MedusaResponse } from "@medusajs/medusa";
```

I didn't really understand what the problem was and just replaced all 
`import { MedusaRequest, MedusaResponse } from "@medusajs/medusa";` 
with 
`import { Request, Response } from "express";`

Also, why do you need to make a build now? First, run the application through `yarn dev` and check that everything is configured correctly

**Comment by quanhavn** [comment]
@siffogh can u help us install project ?

Almost everything in the README file is outdated

**Comment by quanhavn** [comment]
@BloodFalka  
i run BE and got error 
```
error:   Could not resolve module: Workflows. Error: Loaders for module Workflows failed: Invalid URL

error:   Error starting server
Error: Loaders for module Workflows failed: Invalid URL
```

>the backend will not start without the stripe and sendgrid api keys

can i create a mock key? ex: "xxxxx"
Do I need to use the exact key?

**Comment by mangelini** [comment]
Has anyone managed to correctly setup the storefront? After following the instructions above backend seems fine, but on storefront startup I get `Invalid document type: undefined`

**Comment by MohammadBnei** [comment]
Hey, i get the same issue as @mangelini . 404 everywhere, can't do anything

**Comment by MohammadBnei** [comment]
I think that there is an error with sanity being empty. I get access to /cms in the front end part.

**Comment by subflow** [comment]
@MohammadBnei yes i struggle with that also. Im not used to sanity so a guide how to get the data into sanity would be great. I got it kind of running but with alot of errors. Its a bit dissapointing to see medusa website "get up and running in minutes" and i spent a week trying to get this to even run.

**Comment by mindhealer24** [comment]
Still not able to run frontend. Getting Invalid document type: undefined error.

**Comment by lborruto** [comment]
Same here, 404 all over the place..
Would it be possible to update the starter? Thanks,

**Comment by mindhealer24** [comment]
This is the worst of all starter templates. Choose another. This has many bugs which you will discover later if you fixed one.

**Comment by YoussefRabeiii** [comment]
so saaad after all the media and blogs and fancy ads and after all its not even running i tried and i made it work but not quite no functionality i was here for the optimistic UI in the first place since medusa is sadly so slow so i found the commit of the optimistic ui and will review it and leave sadly
